synapse/handlers/room_member.py

```diff
@@ -450,8 +455,37 @@ def _update_membership(
             prev_events_and_hashes=prev_events_and_hashes,
             content=content,
         )
+        logcontext.run_in_background(
+            self._send_merged_user_invites,
+            requester, room_id,
+        )
         defer.returnValue(res)
 
+    @defer.inlineCallbacks
+    def _send_merged_user_invites(self, requester, room_id):
+        profile_alias = "#_profile_" + requester.user.localpart + ":" + self.hs.hostname
+        profile_alias = RoomAlias.from_string(profile_alias)
+        profile_room_id, remote_room_hosts = yield self.lookup_room_alias(profile_alias)
+        if profile_room_id:
+            linked_accounts = yield self.state_handler.get_current_state(
+                room_id=profile_room_id.to_string(),
+                event_type="m.linked_accounts",
+                state_key="",
+            )
+            if not linked_accounts or not linked_accounts.content['all_children']:
+                return
+            for child_id in linked_accounts.content['all_children']:
+                child = UserID.from_string(child_id)
+                if self.hs.is_mine(child) or child_id == requester.user.to_string():
+                    # TODO: Handle auto-invite for local users (not a priority)
+                    continue
+                self.update_membership(
```

Comment:
`update_membership` returns a deferred; you need to `yield` it ior you'll leak logcontexts